docs/Execution.md

```diff
@@ -185,8 +197,39 @@ while (!AllDone) {
 Above, you can see a simple loop that properly drives a single execution context on Windows.
 `OVERLAPPED_ENTRY` objects received from `GetQueuedCompletionStatusEx` are used to get the submission queue entry and then call its completion handler.
 
-In a real application, these completion events may come both from MsQuic and the application itself, therefore, this means **the application must use the same base format for its own submission entries**.
-This is necessary to be able to share the same event queue object.
+Applications may also perform their own IOs alongside those of MsQuic, pointing to the same IOCP, so that the same threads may efficiently service both MsQuic and non-MsQuic IOs together.
+
+The sample above makes the assumption that _all_ dequeued OVERLAPPED_ENTRY values are in fact pointing to `CXPLAT_SQE` objects (which embed the traditional Win32 `OVERLAPPED` structure).
+To simplify mixing MsQuic with non-MsQuic IO, the application may decide to use the same base format as `CXPLAT_SQE` for its own IOs and streamline the event loop as in the above example. This approach will work for all platforms and not only Windows user-mode.
+
+On Windows, if the application has its own structure embedding the `OVERLAPPED` for its own non-MsQuic IOs and cannot align with the same convention as `CXPLAT_SQE`, it may configure MsQuic to use a specific `CompletionKey`
+when posting packets to the IOCP. By inspecting this `CompletionKey`, the application may distinguish MsQuic-based IO (which means it is a `CXPLAT_SQE`), from non-MsQuic IO that may be in any other arbitrary shape.
```

Comment:
> it may configure MsQuic to use a specific `CompletionKey`

But that's not what this PR is actually doing. No where in here are you providing a mechanism to tell the MsQuic library to use a specific key when it creates its own SQEs internally. At best, this API allows the app to configure its own non-zero keys that could be use to distinguished from MsQuic's zero key IOs.